File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
# 1. goal 길이만큼 for문
# 2. 해당 순서(i) 에서 cards1, cards2 첫번째 원소중 있는지 확인
# 3. 각각 없으면 패스
# 4. 둘다없으면 No 리턴
# 5. 끝나면 Yes 리턴
def solution(cards1, cards2, goal):
    answer = ''
    for i in range(len(goal)):
        if cards1:
            if goal[i] == cards1[0]:
                cards1.pop(0)
                continue
                
        if cards2:
            if goal[i] == cards2[0]:       
                cards2.pop(0)
                continue
        answer = "No"
        return answer
    
    answer = "Yes"
    return answer

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

string solution(vector<string> cards1, vector<string> cards2, vector<string> goal) {
    string answer = "";
    vector<string> s = {"a", "b", "c"};
    
    for (string s  : goal){
        if (!cards1.empty()){
            if(cards1[0] == s){
                cards1.erase(cards1.begin());
                continue;
            }
        }
            
        if (!cards2.empty()){
            if(cards2[0] == s){
                cards2.erase(cards2.begin());
                continue;
            }
        }
        // for (string s : cards1){
        //     cout << s;
        // }
        // for (string s : cards2){
        //     cout << s;
        // }
        return "No";
    }
    
    return "Yes";
}




// push_pop(0) 하면 첫번재거 되나? -> 안된다 그럼 매개변수 못오나/  아님 erase(0)도 안됨? -> v.erase(v.begin()) 
 
// 1.goal길이만큼 for문 문자 c 
// 2. c에 해당하는게 cards1[0] 이나 cards2[0] 에 있으면 스킵 없으면 return "NO"
// 3. for문끝나면 yes반환